obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+        msg = 'event has an origin, but time is not populated.  ' +\
+            'This is required!'
+        raise ValueError(msg)
+    # Attempt to cope with possible pre-existing files
+    if not filename:
+        range_list = []
+        for i in range(30):  # Look +/- 30 seconds around origin time
+            range_list.append(i)
+            range_list.append(-1 * i)
+        range_list = range_list[1:]
+        for add_secs in range_list:
+            sfilename = (evtime + add_secs).datetime.strftime('%d-%H%M-%S') +\
+                evtype[0] + '.S' + (evtime + add_secs).\
+                datetime.strftime('%Y%m')
+            if not os.path.isfile(outdir + os.sep + sfilename):
+                sfile = open(outdir + os.sep + sfilename, 'w')
+                break
+            elif overwrite:
+                sfile = open(outdir + os.sep + sfilename, 'w')
+                break
+        else:
+            raise IOError(outdir + os.sep + sfilename +
+                          ' already exists, will not overwrite')
+    else:
+        sfile = open(outdir + os.sep + filename, 'w')
+    # Write the header info.
+    if event.origins[0].latitude:
+        if event.origins[0].latitude not in [float('NaN'), 999]:
+            lat = '{0:.3f}'.format(event.origins[0].latitude)
+        else:
+            lat = ''
+    else:
+        lat = ''
+    if event.origins[0].longitude:
+        if event.origins[0].longitude not in [float('NaN'), 999]:
+            lon = '{0:.3f}'.format(event.origins[0].longitude)
+        else:
+            lon = ''
+    else:
+        lon = ''
+    if event.origins[0].depth:
+        if event.origins[0].depth not in [float('NaN'), 999]:
+            depth = '{0:.1f}'.format(event.origins[0].depth/1000)
+        else:
+            depth = ''
+    else:
+        depth = ''
+    if event.creation_info:
+        try:
+            agency = event.creation_info.get('agency_id')
+            # If there is creation_info this may not raise an error annoyingly
+            if agency is None:
+                agency = ''
+        except AttributeError:
+            agency = ''
+    else:
+        agency = ''
+    if len(agency) > 3:
+        agency = agency[0:3]
+    # Cope with differences in event uncertainty naming
+    if event.origins[0].time_errors:
+        try:
+            timerms = '{0:.1f}'.format(event.origins[0].
+                                       time_errors.Time_Residual_RMS)
+        except AttributeError:
+            timerms = '0.0'
+    else:
+        timerms = '0.0'
+    try:
+        mag_1 = '{0:.1f}'.format(event.magnitudes[0].mag) or ''
+        mag_1_type = _evmagtonor(event.magnitudes[0].magnitude_type) or ''
+        if event.magnitudes[0].creation_info:
+            mag_1_agency = event.magnitudes[0].creation_info.agency_id or ''
+        else:
+            mag_1_agency = ''
+    except IndexError:
+        mag_1 = ''
+        mag_1_type = ''
+        mag_1_agency = ''
+    try:
```

Comment:
These appear to be all the same; you should be able to loop through them as in the read case.